src/Snapshotting/Tests/AggregateSnapshottingTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace EventSauce\EventSourcing\Snapshotting\Tests;
+
+use EventSauce\EventSourcing\AggregateRootId;
+use EventSauce\EventSourcing\AggregateRootRepository;
+use EventSauce\EventSourcing\AggregateRootTestCase;
+use EventSauce\EventSourcing\ConstructingAggregateRootRepository;
+use EventSauce\EventSourcing\MessageDecorator;
+use EventSauce\EventSourcing\MessageDispatcher;
+use EventSauce\EventSourcing\MessageRepository;
+use EventSauce\EventSourcing\Snapshotting\ConstructingAggregateRootRepositoryWithSnapshotting;
+use EventSauce\EventSourcing\Snapshotting\InMemorySnapshotRepository;
+use EventSauce\EventSourcing\Snapshotting\Snapshot;
+
+class AggregateSnapshottingTest extends AggregateRootTestCase
+{
+    /**
+     * @var InMemorySnapshotRepository
+     */
+    protected $snapshotRepository;
+
+    /**
+     * @var ConstructingAggregateRootRepositoryWithSnapshotting
+     */
+    protected $repository;
+
+    /**
+     * @test
+     */
+    public function testing_snapshotting()
+    {
+        $this->given(LightSwitchWasFlipped::off());
+        /** @var LightSwitch $lightSwitch */
+        $lightSwitch = $this->repository->retrieveFromSnapshot($this->aggregateRootId);
+        $this->assertInstanceOf(LightSwitch::class, $lightSwitch);
+        $lightSwitch->turnOn();
+        $this->repository->storeSnapshot($lightSwitch);
+        $this->assertEquals(true, $lightSwitch->state());
```

Comment:
Instead of testing for `true`, use `LightSwitch:ON`? Also applies to assertions further down.